hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleCreateHandler.java

```diff
@@ -250,15 +249,12 @@ public Fees calculateFees(@NonNull final FeeContext feeContext) {
         return scheduleOpsUsage.scheduleCreateUsage(txn, sigUsage, lifetimeSecs);
     }
 
-    private @Nullable Schedule maybeDuplicate(
-            @NonNull final Schedule schedule, @Nullable final List<Schedule> duplicates) {
-        if (duplicates == null) {
+    private @Nullable Schedule maybeDuplicate(@NonNull final Schedule schedule, @Nullable final Schedule duplicate) {
```

Comment:
This should still handle hash collisions, the hash implementation is not robust enough to guarantee no collisions, so it is entirely likely that multiple schedules will match the hash, and need to be checked for duplication.

Reply:
The reason is explained below.